controllers/edgedeployment_controller.go

```diff
@@ -367,3 +381,60 @@ func splitEdgeDevices(edgeDevices []managementv1alpha1.EdgeDevice, splitSize uin
 	}
 	return result
 }
+
+func (r *EdgeDeploymentReconciler) updateLabelsFromSelector(ctx context.Context, edgeDeployment *managementv1alpha1.EdgeDeployment) error {
+	edgeDeploymentCopy := edgeDeployment.DeepCopy()
+	UpdateSelectorLabels(edgeDeploymentCopy)
+	newLabels := edgeDeploymentCopy.Labels
+
+	if (len(newLabels) == 0 && edgeDeployment.Labels == nil) || reflect.DeepEqual(newLabels, edgeDeployment.Labels) {
+		return nil
+	}
+
+	err := r.EdgeDeploymentRepository.Patch(ctx, edgeDeployment, edgeDeploymentCopy)
+	return err
+}
+
+func isSelectorLabel(label string) bool {
+	return strings.HasPrefix(label, selectorLabelPrefix)
+}
+
+func CreateSelectorLabel(label string) string {
+	return selectorLabelPrefix + label
+}
+
+func UpdateSelectorLabels(edgeDeployments ...*managementv1alpha1.EdgeDeployment) {
```

Comment:
This should be private.

Reply:
I use it in tests. Tests are a separate package.

Reply:
I use it in tests. Tests are a separate package.

Reply:
OK, that makes sense in this case.